### DISCUSSION TITLE
A question about interpretation of migration rates (slim vs. msprime)

### DISCUSSION THREAD

**Discussion by mshpak76** [tskit-dev/msprime#2172] [General]
I'm trying to determine if my forward and backward time parameterizations (e.g. Slim vs. msprime) are equivalent, and I'm not certain that they are because it's not clear to me whether the "rate" in the latter refers to the fraction of the source or the destination.

Specifically, if in forward time (Slim code) I have 

    p1.setMigrationRates(p2, 1e-3)

, this means that one in a thousand individuals in p1 in any generation are migrants from population p2.

In reverse time (using msprime syntax), I thought that the equivalent would be

    demography.set_migration_rate(source = "p1", dest = "p2", rate = 1e-3)

However, I'm not sure if this is correct, because doesn't the above actually mean that one one thousandth of p2 to to p1 in forward time, as opposed to a one thousandth of p1 being from p2?

**Comment by petrelharp** [comment]
Good question; a perennial source of confusion. Let's see; perusing the documentation I see that the section on [migration](https://tskit.dev/msprime/docs/latest/demography.html#sec-demography-migration) has some helpful reassuring statements that this is confusing, but maybe unhelpfully it doesn't actually say what migration rates mean. A more concrete statement is in the [definitions](https://tskit.dev/msprime/docs/latest/demography.html#sec-demography-definitions) section, which says

![Screenshot from 2023-04-18 20-38-18](https://user-images.githubusercontent.com/1046249/232960970-44b83068-dd3a-4806-9286-c82f4761f0a5.png)

Ah, but it's easier than that: if "one in a thousand individuals in p1 in any generation are migrants from population p2"; that means that a lineage in p1 has probability 1/1000 of moving to p2 in any generation (i.e., of that individual having it's parent in p2 in that generation). That's exactly the msprime definition of migration rate, with source=p1 and dest=p2. (I think that's right!) So, I think that what you had is equivalent!

ps. I'm going to move this discussion to the msprime repository.

**Reply by mshpak76** [reply]
Thank you for the clarification. Yes, my confusion is/was indeed due to what is meant by the "rate". In forward time, migration rate can mean either the fraction leaving the source or contributing to the destination, which obviously are not equivalent unless the population sizes are the same. Unfortunately, a lot of papers describe models with a "migration rate" without clarifying which of the two they refer to!

**Reply by petrelharp** [reply]
Yes, I agree about the lack of clarity. Here the definition of 'rate' is as in that sentence: "expected number of <events> per unit time".

Also, perhaps this will help clarify things? https://github.com/tskit-dev/msprime/pull/2173

**Comment by mshpak76** [comment]
I think I'm still unclear about how to scale the migration rates with population size. If I have two populations A,B with A twice the size of B and I set a symmetric migration rate (same value, e.g. 0.001 for both), do I get a source-sink effect with respect to A and B?

The reason I ask is that I obtain a very strange result with symmetric migration rates for a simple demographic model. In the script below, I have an ancestral population A of size 2e+4 that splits to B of size 1e+4 (at t=2e+3 generations) when then splits into another population C of the same size at t=1e+3 generations. In the absence of migration, or a very small migration rate, I get, as expected, Fst(A,B)=Fst(A,C) > Fst(B,C).  However, if I introduce a non-trivial symmetric migration rate 1e-3 per generation between all populations, I get Fst(B,C) > Fst(A,B), Fst(A,C). The only explanation I can think of for this is a source-sink effect due to A having a larger population size, though this shouldn't be occurring because (as I understand it) rate 1e-3 means that one in a thousand of A come from B,C, one in a thousand from B come from C,A, etc, so that the source size is corrected for. Is this the case? I can't otherwise account for why Fst(B,C) > Fst(A,B),Fst(A,C) with migration but not in its absence.

My demographic model is given below:

    demography = msprime.Demography()
    demography.add_population(name = "A", initial_size = 20000, initially_active=True)
    demography.add_population(name = "B", initial_size = 10000, initially_active=True)
    demography.add_population(name = "C", initial_size = 10000)
    demography.set_symmetric_migration_rate(["A","B","C"],0.001)I
    demography.add_population_split(time=1000, derived=["C"], ancestral="B")
    demography.add_population_split(time=2000, derived=["B"], ancestral="A")
